gammapy/estimators/map/ts.py

```diff
@@ -745,6 +812,88 @@ def estimate_errn_errp(self, dataset, result):
         )
         return {"norm_errn": flux_errn, "norm_errp": flux_errp}
 
+    def sample_norm(self, result):
+        """Sample different scales to compute profile."""
+
+        norm_err = result["norm_err"]
+        norm = result["norm"]
+        if ~np.isfinite(norm_err) or norm_err == 0:
+            norm_err = 0.1
+        if ~np.isfinite(norm) or norm == 0:
+            norm = 1.0
+        sparse_norms = np.concatenate(
+            (
+                norm + np.linspace(-2.5, 2.5, 51) * norm_err,
+                norm + np.linspace(-10, 10, 21) * norm_err,
+                np.abs(norm) * np.linspace(-10, 10, 21),
+                np.linspace(-10, 10, 21),
+                np.linspace(self.norm.scan_values[0], self.norm.scan_values[-1], 2),
+            )
+        )
+        sparse_norms = np.unique(sparse_norms)
+        if len(sparse_norms) != 109:
+            rand_norms = 20 * np.random.rand(109 - len(sparse_norms)) - 10
+            sparse_norms = np.concatenate((sparse_norms, rand_norms))
```

Comment:
The len of stat_scan should be the shape so it can be stored into a map. But is it not always the case, if norm is divisible by norm_err line 826 and 826 can give the same value.